### ISSUE TITLE
ring_log library shows up previously read log

### ISSUE THREAD

**Issue by maheshsutariya** [menhirsys/ring_log#7]
Hi Menhirsys,

Thank you for writing such a nice library. 

I am trying to use it to store the data in the flash. But I am facing one issue.

I compiled example.c file from
https://github.com/menhirsys/ring_log/blob/master/example.c
for the ESP32.

But when I run this example first time on the fresh device, It shows correct data(3 data) read as below.
one
two
three

But when I restart this device then it shows incorrect data read(4 instead of 3 data) as below.
three
one
two
three

So basically it remembers last read data "three" after restart/power lost. But actually there is only 3 data which was written and not read yet.

I tried to check this problem but couldn't figure out the exact issue.

Will you please check this?

**Comment by edmund-huber** [comment]
Hi Mahesh, in fact, I don't really have the time to work on this library lately. Sorry and good luck!

**Comment by maheshsutariya** [comment]
Ok. I got it working. Thanks. It needed flushing write queue.
